### ISSUE TITLE
SQL SCRUB for constraint checker

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#2043]
Background: https://airtable.com/tblD3oZPLJgGhCmch/viwINAgXA5on9pr2T/recBCBliKS0yfBy8p
RFC: https://github.com/cockroachdb/cockroach/blob/master/docs/RFCS/20171025_scrub_sql_consistency_check_command.md

PRs:
- [sql,parser: Add experimental database check SQL statement](https://github.com/cockroachdb/cockroach/pull/19161)
- [Add SCRUB TABLE ... INDEX and SCRUB TABLE ... INDEX(idxname) syntax](https://github.com/cockroachdb/cockroach/pull/19583)
- [Add SCRUB TABLE ... WITH OPTIONS PHYSICAL syntax](https://github.com/cockroachdb/cockroach/pull/19857)
- [Add SCRUB DATABASE and fix an issue with NULL  #values](https://github.com/cockroachdb/cockroach/pull/19889)
- [Add SCRUB support for checking SQL CHECK and FOREIGN KEY](https://github.com/cockroachdb/cockroach/pull/19960)
- [Add SCRUB checks for physical table integrity](https://github.com/cockroachdb/cockroach/pull/19961), in particular the list of errors that happen may be helpful.
- [Add support for AS OF SYSTEM TIME clause with SCRUB](https://github.com/cockroachdb/cockroach/pull/20368)

**Comment by jseldess** [comment]
@lego, @lhirata is going to start learning about and documenting SCRUB. While you're still here, would you be willing to track down all of the relevant PRs and list them here? Lauren might want to set up time with you as well.

**Comment by lego** [comment]
I updated the list of PRs to gather up all of the ones related to scrub. Unfortunately, many of them don't really document what the command does, other than how to invoke it.

A challenge is that this troubleshooting tool is presenting errors coming from the internals of CockroachDB, and those are definitely prone to change. On that note, the table of errors in the command's RFC is not completely correct, as the errors that we could detect turned out a bit different. I'll update this ASAP.

There is also second RFC (currently just drafted) which expands more on the first RFCs list of errors, and what they may have been caused by or what can be done, https://github.com/cockroachdb/cockroach/pull/20293.
